src/types/promotion.ts

```diff
@@ -1,10 +1,10 @@
 import { RestaurantDetails } from './restaurant';
 
 export enum Sort {
```

Comment:
When is it best to use enum vs interface?

Reply:
Yeah! So an interface is basically a JSON object, and it allows you to name the key and the type of the value:
e.g.
```typescript
interface SomeObject {
  aString: string;
  someOtherProperty: boolean;
  ...
}

// usage
const foo: SomeObject = {
  aString: "any string can go here",
  someOtherProperty: true,
 ... // other properties...
}
```

But an enum on the other hand only allows you take on ONE value at a time. So in this example:
```typescript
enum Sort {
  Default = 'Distance',
  Distance = 'Distance',
  Popularity = 'Popularity',
  Recency = 'Recency',
}

const option: Sort = Sort.Default; // usage
```
So `option` here can only either be `Sort.Default`, or `Sort.Distance`, or `Sort.Popularity`, or `Sort.Recency`. It can't take on more than one value at a time!

Reply:
Lemme know if that makes sense!

Reply:
Oooo I never knew this, thanks so much for explaining!